### DISCUSSION TITLE
v_args wrapper

### DISCUSSION THREAD

**Discussion by shewitt-au** [lark-parser/lark#1471] [Q&A]
I'm hoping that the v_args decorator with the wrapper option is what I'm looking for. I can't find any info on this feature much beyond its existence. If this is though lack of looking on my part I apologise in advance. Any information would be much appreciated. Thanks.

**Comment by erezsh** [comment]
Yes, it's not well documented. wrapper is a callback that accepts `(f, data, children, meta)`, where f is the user callback.

Here is the implementation of `inline=True`, `meta=True`, etc., using "wrapper" functions:

```python
def _vargs_inline(f, _data, children, _meta):
    return f(*children)
def _vargs_meta_inline(f, _data, children, meta):
    return f(meta, *children)
def _vargs_meta(f, _data, children, meta):
    return f(meta, children)
def _vargs_tree(f, data, children, meta):
    return f(Tree(data, children, meta))
```
